src/components/tl/cuda/alltoall/alltoall.c

```diff
@@ -27,6 +27,10 @@ ucc_status_t ucc_tl_cuda_alltoall_init(ucc_base_coll_args_t *coll_args,
         return UCC_ERR_NOT_SUPPORTED;
     }
 
+    if (!ucc_coll_args_is_predefined_dt(&coll_args->args, UCC_RANK_INVALID)) {
```

Comment:
general comment for all cuda and ucp colls:
why not add this function into cuda_task_init() / ucp_task_init() , similar to how you did in nccl?

Reply:
added for cuda colls, for ucp colls it doesn't make sense since tl ucp will perhaps be the only tl with not predefined dt support